src/datanode/src/heartbeat.rs

```diff
@@ -144,6 +144,19 @@ impl HeartbeatTask {
 
         Ok(())
     }
+
+    pub async fn close(&self) -> Result<()> {
+        let running = self.running.clone();
+        if running
+            .compare_exchange(true, false, Ordering::AcqRel, Ordering::Acquire)
+            .is_err()
+        {
+            warn!("Call close heartbeat task multiple times");
+            return Ok(());
```

Comment:
This `return` is unnecessary, it's returned at last.

Reply:
> This `return` is unnecessary, it's returned at last.

Also leading to mismatched types error: 
```
error[E0308]: mismatched types
   --> src/datanode/src/heartbeat.rs:155:13
    |
150 | /         if running
151 | |             .compare_exchange(true, false, Ordering::AcqRel, Ordering::Acquire)
152 | |             .is_err()
153 | |         {
154 | |             warn!("Call close heartbeat task multiple times");
155 | |             Ok(())
    | |             ^^^^^^ expected `()`, found `Result<(), _>`
156 | |         }
    | |_________- expected this to be `()`
    |
    = note: expected unit type `()`
                    found enum `std::result::Result<(), _>`
help: consider using a semicolon here
    |
155 |             Ok(());
    |                   +
help: consider using a semicolon here
    |
156 |         };
    |          +
help: you might have meant to return this value
    |
155 |             return Ok(());
    |             ++++++       +

For more information about this error, try `rustc --explain E0308`.
```

Or change it to the following?
```rust
    pub async fn close(&self) -> Result<()> {
        let running = self.running.clone();
        if running
            .compare_exchange(true, false, Ordering::AcqRel, Ordering::Acquire)
            .is_err()
        {
            warn!("Call close heartbeat task multiple times");
            Ok(())
        } else {
            Ok(())
        }
    }
```

Reply:
> What about:
> 
> ```rust
> let running = self.running.clone();
>         if running
>             .compare_exchange(true, false, Ordering::AcqRel, Ordering::Acquire)
>             .is_err()
>         {
>             warn!("Call close heartbeat task multiple times");
>         } 
>        Ok(())
> ```
👌